ark/analysis/dimensionality_reduction.py

```diff
@@ -78,15 +74,15 @@ def visualize_dimensionality_reduction(cell_data, columns, category, color_map="
         color_map (str):
             Name of MatPlotLib ColorMap used, default is Spectral
         algorithm (str):
-            Name of dimensionality reduction algorithm, default is UMAP
+            Name of dimensionality reduction algorithm, must be UMAP, PCA, or tSNE, default is UMAP
         save_dir (str):
             Directory to save plots, default is None
     """
+
     cell_data = cell_data.dropna()
+    dim_reduction_algos = ["UMAP", "PCA", "tSNE"]
 
-    if algorithm not in ["UMAP", "PCA", "tSNE"]:
-        raise ValueError(f"The algorithm specified must be one of the following: "
-                         f"{['UMAP', 'PCA', 'tSNE']}")
+    misc_utils.verify_in_list(algorithm=algorithm, dim_reduction_algos=dim_reduction_algos)
```

Comment:
I'm on the fence about this one. It's nice only having to provide 2 args instead of 4, especially when the extra args were basically just the string version of the variable. However, the downside is I think it's slightly more confusing to figure out what this function is doing, and it doesn't give us the flexibility to provide a different name for the variables.

Reply:
For example, if it was instead 
```
verify_in_list(values=algorithm, valid_values=dim_reduction_algos, values_names='algorithm', valid_names='dimensionality reduction algorithms')
```

It's longer and more annoying to write, but easier to read and slightly more descriptive error message. I'm fine with whatever you and @vacuousplanet think is better

Reply:
I'm also on the fence about this one. @vacuousplanet what do you think?

Reply:
Yeah, I'm also on the fence. @ackagel what do you say?

Reply:
You could allow a custom error string to be passed in as a kwarg:
```
err_str = '%s doesn't accept the value: %s.  Valid dimensionality reduction algorithms are: %s'
misc_utils.verify_in_list(algorithm=algorithm, dim_reduction_algos=dim_reduction_algos, err_str=err_str)
```
Then, just check for provided err_str, and otherwise use the default, in `verify_in_list`.

One extra way you could go with this, is for a provided `err_str` to use all of the other provided kwargs in order.  The default `err_str` should stay tho; it looks like it works pretty well for most cases.

Reply:
That made me realize something: we don't *have* to name the args the same as the variable. 

For example:
```
verify_in_list(algorithm=algorithm, dimesionality_reduction_algorithms=dim_reduction_algos)
```

This would work with Alex's current version, and allows us to basically make custom error messages, as long as we're okay with any new stuff being a single string. We could optionally split the arg_names based on _ before putting them into the error string for extra readability. 

Or, we could make it a dict, then we could have the keys be strings and values the appropriate values. But it wouldn't be ordered, so I think that would cause more of a headache

Reply:
@ngreenwald I'm not a big fan of the dict method personally because of the reason you mentioned (unordered) and also because I think it's just a little bit more work than necessary (`**kwargs` basically accomplish the same thing and don't require dicts to be passed). As you mention, having flexible `**kwargs` also allows the variable names passed to be different.

@ackagel I do like the idea of custom error messages, but I think the original idea of this function was to eliminate the need of the user to format/print the error message on their end. The concern is that if we'll require a ton of custom error messages, it could defeat the purpose of having an easy-to-use function in `misc_utils` which allows a simple list in list/set equality check and an appropriate error message displayed if necessary.

There are definitely a lot of tradeoffs we'll have to make here, but unless you guys have extra concerns, I'll move forward with Noah's most recent suggestion of leaving the `**kwargs` as is and allowing the user to specify the arg names.